moveit_kinematics/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/NearestNeighborsGNAT.h

```diff
@@ -551,7 +551,9 @@ class NearestNeighborsGNAT : public NearestNeighbors<_T>
         std::vector<int> permutation(children_.size());
         for (unsigned int i = 0; i < permutation.size(); ++i)
           permutation[i] = i;
-        std::random_shuffle(permutation.begin(), permutation.end());
+        std::random_device rng;
```

Comment:
Random device and generator could be member variables that can be reused. Also, are these relevant for any test? Otherwise, we should consider allowing to use a constant seed to prevent flaky CI (however, `std::random_shuffle` doesn't use a seed either).